src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellItemRenderer.cs

```diff
@@ -406,12 +438,43 @@ protected virtual void OnTabReselected(ShellSection shellSection)
 
 		protected virtual void SetupMenu(IMenu menu, int maxBottomItems, ShellItem shellItem)
 		{
+			if (_menuSetup)
+			{
+				if (_updateMenuItemIcon != null &&
+					_updateMenuItemSource != null &&
+					_bottomView.IsAlive() &&
+					_bottomView?.IsAttachedToWindow == true)
+				{
+					var menuItem = _updateMenuItemIcon;
+					var shellSection = _updateMenuItemSource;
+					_updateMenuItemSource = null;
+					_updateMenuItemIcon = null;
+
+					UpdateShellSectionIcon(shellSection, menuItem);
+					return;
+				}
+				else if (_updateMenuItemTitle != null &&
+					_updateMenuItemSource != null &&
+					_bottomView.IsAlive() &&
+					_bottomView?.IsAttachedToWindow == true)
```

Comment:
What is the diff between the following to if statements:

Previous one:
```csharp
				if (_updateMenuItemIcon != null &&
					_updateMenuItemSource != null &&
					_bottomView.IsAlive() &&
					_bottomView?.IsAttachedToWindow == true)
```
This:
```csharp
				else if (_updateMenuItemTitle != null &&
					_updateMenuItemSource != null &&
					_bottomView.IsAlive() &&
					_bottomView?.IsAttachedToWindow == true)
```
The diff is just the first part of the clause, I'm not sure how expensive are the method calls, but a null check is very cheap, you can merge both to something similar to:
```csharp
if (updateMenuItemSource != null &&
 	_bottomView?.IsAlive() &&
 	_bottomView?.IsAttachedToWindow == true) {
    if (_updateMenuItemSource != null) {
      // update here
    } else if (_updateMenuItemTitle != null) {
     // update here
   }
}
```

Reply:
Yea it's probably a micro-optimization but I just didn't want any of this code to run if both `_updateMenuItemIcon ` and `_updateMenuItemTitle ` are null. 

I shifted it around a bit to have left stuff but still left a check in for those two so that the if statement can get short circuited before calling things on `BottomView` which will traverse the bridge